src/transformers/integrations/deepspeed.py

```diff
@@ -289,6 +290,19 @@ def is_deepspeed_zero3_enabled():
         return False
 
 
+def is_deepspeed_zero3_enabled_accelerate():
+    """
+    Will check if the user has enabled ZeRO 3 with accelerate. This is useful
+    when checking if the environment is configured for it before
+    the deepspeed config reference has been made.
+    """
+    accelerate_zero_stage = int(os.environ.get("ACCELERATE_DEEPSPEED_ZERO_STAGE", -1))
+    accelerate_zero_init = os.environ.get("ACCELERATE_DEEPSPEED_ZERO3_INIT", "0")
```

Comment:
`ACCELERATE_DEEPSPEED_ZERO3_INIT` is either `true` or `false`. The below check `accelerate_zero_init != "0"` would evaluate to True even if `ACCELERATE_DEEPSPEED_ZERO3_INIT` is `false`.